### DISCUSSION TITLE
Where to put JS snippets that used to reside in CRA's index.html head?

### DISCUSSION THREAD

**Discussion by websocket98765** [vercel/next.js#18180] [Help] [answered]
In Create React App, I used to have a snippet like below in my index.html head section. It instantiates an NPM lib to use settings for my website. But I can't include `<script>` tags in the NextJS header. 

Where should this type of snippet go in NextJS?  

```js
<script type="text/javascript">
    window.ACME_CONFIG = {
      hosts: [
        {
          domain: 'example.com',
          foo: bar
        }
      ]
    }
</script>
```

What I tried: I removed the HTML tags, added a 'window is not undefined' conditional so the server doesn't error, & then I tried placing it in my `next/head` component, `_app.js`, & `pages/index.js` but haven't found a spot where it works yet. Oddly, I've seen it work noce or twice, but without changing any code, it would then fail again consistently after a hard refresh :(

**Comment by jamesmosier** [comment, marked as answer]
You could do something similar to the [with-google-analytics](https://github.com/vercel/next.js/blob/58c161c99bc400ab2c1553afc476cdddeceffa46/examples/with-google-analytics/pages/_document.js#L15-L27) example and create a `_document.js` file to place it in:

```js
import Document, { Html, Head, Main, NextScript } from 'next/document'

export default class MyDocument extends Document {
  render() {
    return (
      <Html>
        <Head>
          <script
            dangerouslySetInnerHTML={{
              __html: `
                window.ACME_CONFIG = {
                  hosts: [
                    {
                      domain: 'example.com',
                      foo: bar
                    }
                  ]
                }
          `,
            }}
          />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}
```

---

Or you *should* be able to do the same thing with `next/head`:

```js
import Head from 'next/head'

<Head>
  <script
    dangerouslySetInnerHTML={{
      __html: `
        window.ACME_CONFIG = {
          hosts: [
            {
              domain: 'example.com',
              foo: bar
            }
          ]
        }
  `,
    }}
  />
</Head>
```

**Reply by websocket98765** [reply]
I tried the second one and it worked great! It's nice & clean to be able to do it in my custom head.

Thank you!
